cypress/integration/membership.spec.js

```diff
@@ -0,0 +1,62 @@
+import faker from 'faker'
+
+describe('Membership Spec', () => {
+  beforeEach(() => {
+    cy.request('/cypress_rails_reset_state')
+  })
+
+  describe('happy', () => {
+    it('renders donation widget', () => {
+      cy.visit('/test/widget/membership')
+      cy.contains('PAY MONTHLY DUES', { matchCase: false })
+    })
+  })
+
+  describe('edge cases', () => {
+    it('creates a membership if user has a stripe_customer without any payment method attached', () => {
+      // setup stripe mocks
+      cy.appScenario('basic')
+      cy.forceLogin({ email: 'empty_stripe_id@example.com' })
+
+      cy.visit('/test/widget/membership')
+
+      // select amount
+      cy.contains('$10').click()
+      cy.get('button')
+        .contains('PAY MONTHLY DUES', { matchCase: false })
+        .click()
+
+      // fill address
+      cy.contains('PAYING $10', { matchCase: false }).click()
+
+      const zipCode = faker.address.zipCode()
+      cy.get("input[name='address']").type(faker.address.streetAddress())
```

Comment:
If you use instead https://testing-library.com/docs/cypress-testing-library/intro/ we will have a unified "language" on the test side as you will be using rtl queries instead.

_Highly opinionated comment, feel free to proceed as you believe convenient_

Reply:
Yeah I think we can add this later.